chia/types/blockchain_format/coin.py

```diff
@@ -53,11 +53,11 @@ def __bytes__(self) -> bytes:  # pylint: disable=E0308
         assert False
 
 
-def hash_coin_list(coin_list: List[Coin]) -> bytes32:
-    coin_list.sort(key=lambda x: x.name_str, reverse=True)
+def hash_coin_ids(coin_ids: List[bytes32]) -> bytes32:
+    coin_ids.sort(reverse=True)
     buffer = bytearray()
 
-    for coin in coin_list:
-        buffer.extend(coin.name())
+    for name in coin_ids:
+        buffer.extend(name)
 
     return std_hash(buffer)
```

Comment:
oh. it seems like a mistake to me to always construct a new `bytes` object from the input. but also, does a list of `bytes` even support being converted to `bytes` and are they just concatenated? I wouldn't have thought a list would be allowed as input.

Reply:
Bother, I missed a layer for some of my comments.  Sorry.  Ok, so `std_hash(b for coin_id in coin_ids for b in coin_id)` to flatten via a generator.  Or `std_hash(itertools.chain(*coin_ids))`.  And I stand by my claim that this may or may not be better.  `:]`